File: amirmp/rpc.mjs
import { Api, JsonRpc, JsSignatureProvider } from "ineryjs";
import dotenv from "dotenv";
dotenv.config();

const privateKey = process.env.PRIVATE_KEY;
const signature = new JsSignatureProvider([privateKey]);

const json_rpc = new JsonRpc(process.env.URL_NODE);

const account = "amirmp";
const actor = "amirmp";


const pushApi = new Api({
  rpc: json_rpc,
  signatureProvider: signature,
});

async function createData(id, user, data){
  try{
      const tx = await pushApi.transact({
          actions:[
              {
                  account,
                  name:"create",
                  authorization:[
                      {
                          actor,
                          permission:"active",
                      }
                  ],
                  data:{
                      id, user, data
                  }
              }
          ]
      },{broadcast:true,sign:true})

      
    console.log(tx)
    console.log(tx.processed.action_traces[0].console)
  } catch (error) {
    console.log(error);
  }
}

async function createTx(id, user, data){
  try{
      const tx = await pushApi.transact({
          actions:[
              {
                  account,
                  name:"create",
                  authorization:[
                      {
                          actor,
                          permission:"active",
                      }
                  ],
                  data:{
                      id, user, data
                  }
              }
          ]
      },{broadcast:true,sign:true})

      
    console.log(tx)
    console.log(tx.processed.action_traces[0].console)
  } catch (error) {
    console.log(error);
  }
}


const main = async () => {
  await createData(5, account, "newdata");
  await createTx(5, account, "createtx");

};

main();

File: suhar/rpc.py
from eospy.cleos import Cleos
import os
url = os.environ.get('NODE_URL')
ce = Cleos(url)
# Inery Smart Contract Account to Call
account = os.environ.get('INERY_ACCOUNT')
actor = 'suhar'

# private key
private_key = os.environ.get('PRIVATE_KEY')


def create(id, user, data):
    try:
        # create new transaction and sign it
        trx = {"actions":[{"account":account,"name":"create","authorization":[{"actor":actor,"permission":"active"}],"data":{"id":id, "user":user, "data":data}}]}
        opts = {"broadcast": True, "sign": True}
        resp = ce.push_transaction(trx, opts, [private_key])
        
        print(resp)
        print(resp['processed']['action_traces'][0]['console'])
    except Exception as e:
        print(e)


create(10, account, "Create new Data via JSON RPC")
